tools/LogCollector/source/update_mgmt_health_check.py

```diff
@@ -0,0 +1,266 @@
+        if "*" in endpoint and workspace is not None:
+            new_endpoint = endpoint.replace("*", workspace)
+        elif "*" not in endpoint:
+            new_endpoint = endpoint
+
+        if new_endpoint is not None:
+            try:
+                response = sock.connect_ex((new_endpoint, 443))
+
+                if response == 0:
+                    output.append(new_endpoint + ": success\n")
+                else:
+                    output.append(new_endpoint + ": failure\n")
+
+            except Exception as ex:
+                output.append(new_endpoint + ": failure (this is normal if region is in Fairfax) \n")
+                output.append(str(ex) + "\n")
+    
+    output.append("\n")
+
+    for endpoint in ff_endpoints:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+        new_endpoint = None
+        
+        if "*" in endpoint and workspace is not None:
+            new_endpoint = endpoint.replace("*", workspace)
+        elif "*" not in endpoint:
+            new_endpoint = endpoint
+
+        if new_endpoint is not None:
+            try:
+                response = sock.connect_ex((new_endpoint, 443))
+
+                if response == 0:
+                    output.append(new_endpoint + ": success\n")
+                else:
+                    output.append(new_endpoint + ": failure\n")
+
+            except Exception as ex:
+                output.append(new_endpoint + ": failure (this is normal if the region is not Fairfax).\n")
+                output.append(str(ex) + "\n")
+        
+    return "".join(output)
+
+def check_oms_agent_installed():
+    oms_agent_dir = "/var/opt/microsoft/omsagent"
+    oms_agent_log = "/var/opt/microsoft/omsagent/log/omsagent.log"
+
+    agent_status = ["Oms Agent: \n"]
+
+    if os.path.isdir(oms_agent_dir):
+        # Check for omsagent.log
+        if os.path.isfile(oms_agent_log):
+            agent_status.append("omsagent.log exists. Oms Agent looks to be installed. \n")
+        else:
+            agent_status.append("File /var/opt/microsoft/omsagent/log/omsagent.log does not exist. Is the omsagent installed? \n")    
+        
+        # Check for multihoming of workspaces
+        directories = []
+        potential_workspaces = []
+
+        for (dirpath, dirnames, filenames) in walk(oms_agent_dir):
+            directories.extend(dirnames)
+            break # Get the top level of directories
+
+        for directory in directories:
+            if len(directory) >= 32:
+                potential_workspaces.append(directory)
+
+        if len(potential_workspaces) > 1:
+            agent_status.append("OMS Agent may be multihomed. Potential workspaces: " + str(potential_workspaces) + "\n")
+
+    else:
+        agent_status.append("Directory /var/opt/microsoft/omsagent/ does not exist. Is the omsagent installed? \n")
+
+    return "".join(agent_status)
+
+def check_hybrid_worker_running():
+    output = os.popen("ps aux | grep worker").read()
```

Comment:
check if process is running can be converted to a method and we can reuse by passing a string for doing the grep